python/tests/xir/test_parser.py

```diff
@@ -63,6 +63,32 @@ def test_options_lists(self, key, val):
         assert key in irprog.options
         assert irprog.options[key] == val
 
+    @pytest.mark.parametrize(
+        "script, adjoint",
+        [
+            ("adjoint ry(2.4) | [2];", True),
+            ("ry(2.4) | [2];", False),
+        ],
+    )
+    def test_adjoint_modifier(self, script, adjoint):
+        """Test that adjoint modifier for gate statments works correctly"""
+        irprog = parse_script(script)
+
+        assert irprog.statements[0].is_adjoint is adjoint
+
+    @pytest.mark.parametrize(
+        "script, ctrl_wires",
+        [
+            ("ctrl[0, 1] ry(2.4) | [2];", (0, 1)),
+            ("ry(2.4) | [2];", None),
+        ],
+    )
+    def test_ctrl_modifier(self, script, ctrl_wires):
+        """Test that no ctrl modifier for gate statments works correctly"""
```

Comment:
I think are a couple of typos here.  Also, we can make the comment more precise:
```suggestion
        """Test that control wires modifier for gate statements correctly sets
        the ``Statement.ctrl_wires`` property.
        """
```